qiskit/test/mock/fake_backend.py

```diff
@@ -38,6 +52,296 @@ def __init__(self, token="123456", url="https://"):
+            list will be returned. If properties are missing for a qubit this
+            can be ``None``.
+        """
+        if isinstance(qubit, int):  # type: ignore[unreachable]
+            return self._qubit_properties.get(qubit)
+        if isinstance(qubit, List):
+            return [self._qubit_properties.get(q) for q in qubit]
+        return None
+
+    def run(self, run_input, **options):
+        """Run on the fake backend using a simulator.
+
+        This method runs circuit jobs (an individual or a list of QuantumCircuit
+        ) and pulse jobs (an individual or a list of Schedule or ScheduleBlock)
+        using BasicAer or Aer simulator and returns a
+        :class:`~qiskit.providers.Job` object.
+
+        If qiskit-aer is installed, jobs will be run using AerSimulator with
+        noise model of the fake backend. Otherwise, jobs will be run using
+        BasicAer simulator without noise.
+
+        Currently noisy simulation of a pulse job is not supported yet in
+        FakeBackendV2.
+
+        Args:
+            run_input (QuantumCircuit or Schedule or ScheduleBlock or list): An
+                individual or a list of
+                :class:`~qiskit.circuits.QuantumCircuit,
+                :class:`~qiskit.pulse.ScheduleBlock`, or
+                :class:`~qiskit.pulse.Schedule` objects to run on the backend.
+            options: Any kwarg options to pass to the backend for running the
+                config. If a key is also present in the options
+                attribute/object then the expectation is that the value
+                specified will be used instead of what's set in the options
+                object.
+        Returns:
+            Job: The job object for the run
+        Raises:
+            QiskitError: If a pulse job is supplied and qiskit-aer is not
+            installed.
+        """
+        circuits = run_input
+        pulse_job = None
+        if isinstance(circuits, (pulse.Schedule, pulse.ScheduleBlock)):
+            pulse_job = True
+        elif isinstance(circuits, circuit.QuantumCircuit):
+            pulse_job = False
+        elif isinstance(circuits, list):
+            if circuits:
+                if all(isinstance(x, (pulse.Schedule, pulse.ScheduleBlock)) for x in circuits):
+                    pulse_job = True
+                elif all(isinstance(x, circuit.QuantumCircuit) for x in circuits):
+                    pulse_job = False
+        if pulse_job is None:
+            raise QiskitError(
+                "Invalid input object %s, must be either a "
+                "QuantumCircuit, Schedule, or a list of either" % circuits
+            )
+        if _optionals.HAS_AER:
+            from qiskit.providers import aer
+
+            if pulse_job:
+                raise QiskitError("Pulse simulation is currently not supported for V2 backends.")
+            sim = aer.Aer.get_backend("qasm_simulator")
```

Comment:
One thing here is that this will suffer from #7500. Can we add:

```suggestion
            sim = aer.Aer.get_backend("qasm_simulator")
            sim._options = self._options
```

(which is all that's required to fix #7500 I think for backendV1 fake backends)

We'll also need to do this for the basic aer path.